src/main/kotlin/uk/gov/dluhc/printapi/client/ElectoralRegistrationOfficeManagementApiClient.kt

```diff
@@ -21,43 +22,56 @@ class ElectoralRegistrationOfficeManagementApiClient(
     private val eroMapper: EroManagementApiEroDtoMapper
 ) {
 
+    /**
+     * Calls the `ero-management-api` to return a [ElectoralRegistrationOfficeResponse] for the specified eroId.
+     *
+     * @param eroId the ID of the ERO to return
+     * @return a [ElectoralRegistrationOfficeResponse] for the ERO
+     * @throws [ElectoralRegistrationOfficeManagementApiException] concrete implementation if the API returns an error
+     */
+    fun getElectoralRegistrationOffice(eroId: String): ElectoralRegistrationOfficeResponse =
+        eroManagementWebClient
+            .get()
+            .uri("/eros/{eroId}", eroId)
+            .retrieve()
+            .bodyToMono(ElectoralRegistrationOfficeResponse::class.java)
+            .onErrorResume { ex -> handleException(ex, "eroId = $eroId") }
+            .block()!!
+
     /**
      * Calls the `ero-management-api` to return a [EroManagementApiEroDto] for the specified gssCode.
      *
      * @param gssCode the gssCode of the localAuthority for the ERO to be returned
      * @return an [EroManagementApiEroDto] for the ERO
      * @throws [ElectoralRegistrationOfficeManagementApiException] concrete implementation if the API returns an error
      */
-    fun getElectoralRegistrationOffice(gssCode: String): EroManagementApiEroDto {
+    fun getElectoralRegistrationOffices(gssCode: String): EroManagementApiEroDto {
         val response = eroManagementWebClient
             .get()
             .uri("/eros?gssCode=$gssCode")
             .retrieve()
             .bodyToMono(ElectoralRegistrationOfficesResponse::class.java)
-            .onErrorResume { ex -> handleException(ex, gssCode) }
+            .onErrorResume { ex -> handleException(ex, "gssCode = $gssCode") }
             .block()!!
 
         if (response.eros.isEmpty()) {
-            throw ElectoralRegistrationOfficeNotFoundException(gssCode)
+            throw ElectoralRegistrationOfficeNotFoundException("ERO not found for gssCode = $gssCode")
```

Comment:
@Neil-Massey . how about just passing in the search criteria only to the NotFoundException and let it deal with the prefix?

<pre>
throw ElectoralRegistrationOfficeNotFoundException(<b>"gssCode = $gssCode"</b>)

class ElectoralRegistrationOfficeNotFoundException(<b>searchCriteria</b>: String) :
    ElectoralRegistrationOfficeManagementApiException(<b>"ERO not found for $searchCriteria"</b>)
</pre>

Reply:
I was in two minds - the `ElectoralRegistrationOfficeGeneralException` was taking full message so went with same approach:
```
abstract class ElectoralRegistrationOfficeManagementApiException(message: String) : RuntimeException(message)

class ElectoralRegistrationOfficeNotFoundException(message: String) :
    ElectoralRegistrationOfficeManagementApiException(message)

class ElectoralRegistrationOfficeGeneralException(message: String) :
    ElectoralRegistrationOfficeManagementApiException(message)
```

Splitting the message and doing string manipulation as you suggested seemed a bit strange (I started with that approach).  But repeating the message at each call site seems wrong too.  Will review on Monday.

Reply:
Thanks, @Neil-Massey , for addressing it.